tests/unit/backend/wmg/api/test_v1.py

```diff
@@ -515,7 +515,7 @@ def test__query_request_with_filter_dims__returns_valid_filter_dims(
     @patch("backend.wmg.api.v1.gene_term_label")
     @patch("backend.wmg.api.v1.ontology_term_label")
     @patch("backend.wmg.api.v1.load_snapshot")
-    def test__query_request_with_filter_dims__returns_elided_filter_dims(
+    def test__query_request_with_filter_dims__secondary_filters_dont_filter_themselves(
```

Comment:
⚠️ I think we will need to update tests more substantially to properly test the new logic. Note that the cube contains the cross-product of all dimension values. With self-filtering removed, the secondary filter dim values in the result are actually no different than if we had specified no secondary filtering at all. So this test can longer faithfully verify that secondary filter dims are being elided. We'll need a more "interesting" cube.  E.g., all rows where disease=X have only ethnicity=Y, .  Then we can filter on disease X and assert that 1) ethnicity secondary dim values are correctly elided and 2) disease secondary values contain all diseases.  These two assertions are better off as  separate test methods.  Let me get back to you on suggestions for how to setup the cube for such tests.

Reply:
The [create_temp_wmg_snapshot](https://github.com/chanzuckerberg/single-cell-data-portal/blob/7438d2de77ee7bd345e1fc576347bc5d70a3da99/tests/unit/backend/wmg/api/test_v1.py#L56) method  has an keyword param `exclude_logical_coord_fn`, which takes a function that allows one to prune rows from what would otherwise be a cube that contains one row for _every_ combination of dimension values (i.e. the rows are the cross-product all dimensions' values).  This _could_ be used to specify a function that to create the necessary cube. It's not a beautiful solution for cube data setup, but it's serviceable. We might want to put more effort in writing test fixture code for generating cubes that have specific data requirements. I can help further on this, if asked!